securedrop_client/logic.py

```diff
@@ -339,6 +343,13 @@ def on_synced(self, result):
                                          remote_submissions,
                                          remote_replies)
 
+            # Add source's keys to the GPG keyring
+            for source in remote_sources:
+                if source.key and source.key.get('type', None) == 'PGP':
+                    pub_key = source.key.get('public', None)
+                    if pub_key:
+                        self.gpg.import_key(source.uuid, pub_key)
```

Comment:
I think we shouldn't rely on the client always having the same key as the server because at some point in the future, we could rotate keys. It would be nice to do this pretty and async, but for now this is fine. I can look in to moving the logic to the other function.